thompson_sampling/thompson_strategy.py

```diff
@@ -78,3 +89,51 @@ def mean(self):
     @property
     def selections(self):
         return self.num_selections
+
+
+class GaussianProcessesStrategy(ThompsonStrategy):
+    def __init__(
+        self,
+        num_parameters: int,
+        interval: "np.ndarray | list[np.ndarray]",
+        beta: float,
+    ):
+        super().__init__()
+        self.num_parameters = num_parameters
+        if isinstance(interval, list):
+            if len(interval) != self.num_parameters:
+                raise ValueError(
+                    "When passing a list the length should be the same as num_parameters"
+                )
+            else:
+                self.mesh = np.meshgrid(*interval)
+
+        elif isinstance(interval, np.ndarray):
+            self.mesh = np.meshgrid(*[interval for _ in range(self.num_parameters)])
```

Comment:
the interval could be different for each parameter.

Reply:
You can pass the interval either as a single `np.ndarray`, giving the same interval for all the parameters. Or, you can pass a list of `np.ndarray` of length self.num_parameters, which could all be different.

Reply:
ok, something like *([interval] * num_parameters)

Reply:
fixed in 94ba55000419022918e29b985e505e5bc109131b